netmiko/utilities.py

```diff
@@ -344,6 +352,31 @@ def get_structured_data(raw_output, platform=None, command=None, template=None):
         )
 
 
+def get_structured_data_ttp(raw_output, template=None):
+    """
+    Convert raw CLI output to structured data using TTP template.
+
+    You can use a straight TextFSM file i.e. specify "template"
+    """
+    if not TTP_INSTALLED:
+        msg = "\nTTP is not installed. Please PIP install ttp:\n" "pip install ttp\n"
+        raise ValueError(msg)
+
+    if os.path.exists(template):
+        with open(template) as f:
```

Comment:
No need to open TTP template and read it , better pass it on to TTP as is, TTP will use  [load_files](https://github.com/dmulyalin/ttp/blob/06794d8226bf406b126637208690051e0068c0a1/ttp/utils/loaders.py#L6) function to load it checking if its an OS path to file, OS path to directory of files or a text string.